messaging/src/main/java/org/axonframework/commandhandling/gateway/AbstractRetryScheduler.java

```diff
@@ -150,6 +195,43 @@ public B retryExecutor(ScheduledExecutorService retryExecutor) {
             return (B) this;
         }
 
+        /**
+         * Sets the {@link Predicate} used for testing whether failure is transient (returns {@code false}) or not (returns {@code true}).
+         * <p/>
+         * Non-transient failures are never retried.
+         * <p/>
+         * If configured, {@code nonTransientFailures} list is ignored.
+         *
+         * @param nonTransientFailurePredicate a {@link Predicate} (accepting a {@link Throwable} parameter) used for testing transiency of provided failure
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public B nonTransientFailurePredicate(Predicate<Throwable> nonTransientFailurePredicate) {
```

Comment:
Binding to `Throwable` as the parameter of the `Predicate`s while technically correct and most powerful, is quite user-unfriendly as every passed predicate would have to check the type and cast the exception. It would be great if there was another builder method that would allow the users to register more specific predicates for types, for example, a `Predicate<CommandExecutionException>` and get that parameter already cast in the predicate. 

Since there is not best solution on what to bind the generic in the base `Predicate` variable, this can be solved by saving the type somewhere else.

One possible solution might be to provide an additional builder method that would have a class object as the first parameter and the Predicate with correct generic as the second parameter. That way, this builder can wrap the user predicate in a new, type-checking and delegating one that is `Predicate<Throwable>` but is able to check the type if it matches the provided `Class` object, casts the `Throwable` and invokes the user predicate. Of course, it would add that new predicate to the predicate chain or a list before building the final predicate/list, depending on the implementation.

This whole point is to reduce unnecessary bloat or plumbing code and reduce the size of those user-defined predicates/filters, by providing a more user-friendly API and wrapping the most common behaviour of these predicates.

The Builder's API would be something similar to Reactor's [this doOnError](https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Flux.html#doOnError-java.lang.Class-java.util.function.Consumer-) variant but instead of recovering, it would apply the predicate, otherwise, ignore that predicate.